libmeepgeom/meepgeom.cpp

```diff
@@ -23,39 +23,55 @@ namespace meep_geom {
 /***************************************************************/
 /* global variables for default material                       */
 /***************************************************************/
-vector3 ones={1.0,1.0,1.0};
-vector3 zeroes={0.0,0.0,0.0};
-
-//susceptibility_list empty_sus_list{ .items=0, .num_items=0 };
-susceptibility_list empty_sus_list{ 0, 0 };
-
-medium_struct vacuum_medium{
- .epsilon_diag        = ones,
- .epsilon_offdiag     = zeroes,
- .mu_diag             = ones, 
- .mu_offdiag          = zeroes,
- .E_susceptibilities  = empty_sus_list,
- .H_susceptibilities  = empty_sus_list,
- .E_chi2_diag         = zeroes,
- .E_chi3_diag         = zeroes,
- .H_chi2_diag         = zeroes,
- .H_chi3_diag         = zeroes,
- .D_conductivity_diag = zeroes,
- .B_conductivity_diag = zeroes
-}; 
-
-material_data vacuum_material_data{
- material_data::MEDIUM, 0, 0, &vacuum_medium
-};
+material_type vacuum;
+material_type air;
+medium_struct vacuum_medium;
+material_data vacuum_material_data;
+bool materials_initialized = false;
+
+void initialize_materials()
```

Comment:
Static structure initializations can be done even in 1990 ANSI C.  You just can't use named fields.  i.e.
```c
const medium_struct vacuum_medium = {{1,1,1},{0,0,0},{1,1,1},{0,0,0}};
const material_data vacuum_material = {MEDIUM,NULL,NULL,&vacuum_medium};
const material_type vacuum = {(void*) &vacuum_material};
```
(Note that the unspecified extra fields in `vacuum_medium` are initialized to 0, per the C/C++ standard.)